docs/advanced-solidus/payments-and-refunds.mdx

```diff
@@ -571,6 +571,12 @@ provided by Solidus for payment sources:
   end
 ```
 
+We've introduced a new checkout parameter that Solidus needs to be made aware of, we can follow [the guide on introducing custom model attributes](https://guides.solidus.io/developers/customizations/customizing-attributes#allowing-custom-attributes-to-be-saved) and add the following:
+
+```ruby title="config/initializers/spree.rb
+Spree::PermittedAttributes.checkout_payment_attributes << :auth_token
+```
```

Comment:
Hi @alecvn ! I tested this on https://github.com/gsmendoza/rails_6161_store/tree/edgeguides-50 and found that we need to add `auth_token` to `source_attributes` instead, like this:

```ruby title="config/initializers/spree.rb
Spree::PermittedAttributes.source_attributes << :auth_token
```

Can you confirm? This is how I manually tested it:

Given I have cloned git@github.com:gsmendoza/rails_6161_store.git
And I have checked out the branch https://github.com/gsmendoza/rails_6161_store/tree/edgeguides-50
And I have created the app's development database, migrated the database, and seeded it
And I am now running the app
And I am now checking out an order
And I am now in the Payment step
When I select Solidus Pay
And I enter any value for the SolidusPay Auth Token
And I click "Save and Continue"
Then I should see the payment step succeed
And I should see the Checkout page

With this in mind, can I recommend the following change? This also moves the `auth_token` tip next to the payment form partial step, where I think it's more relevant:

---
The first partial we'll implement is used by Solidus when displaying the SolidusPay payment form in
the checkout flow. We will just ask users for their SolidusPay auth token:

```markup title="app/views/spree/checkout/payment/:solidus:pay.html.erb"
<% param_prefix = "payment_source[#{payment_method.id}]" %>

<div class="field field-required">
  <%= label_tag "auth_token_#{payment_method.id}", 'SolidusPay Auth Token' %>
  <%= text_field_tag "#{param_prefix}[auth_token]", nil, { id: "auth_token_#{payment_method.id}" } %>
</div>
```

We'll also need to add `auth_token` as a permitted source attribute in order for
it to get accepted by `Spree::CheckoutController`:

```ruby title="config/initializers/spree.rb
Spree::PermittedAttributes.source_attributes << :auth_token
```

When users fill this form during checkout, Solidus will create a new `SolidusPayAccount` payment
source with the auth token provided by the user. In general, Solidus will copy all
the `#{param_prefix}[...]` attributes to the payment source, so you can add more columns to the
payment source model and Solidus will set them all as long as your field names are structured
properly.

---

Reply:
Hi @gsmendoza

Thanks for double checking this, I haven't had the chance to update my repo - I ended up using one of the existing allowed attributes `verification_value` in the meantime.

I will have a look at this after work, confirm, and then update my PR accordingly.